contracts/strategies/StrategymUSDCurve.sol

```diff
@@ -0,0 +1,288 @@
+pragma solidity ^0.5.17;
+
+import "@openzeppelinV2/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelinV2/contracts/math/SafeMath.sol";
+import "@openzeppelinV2/contracts/utils/Address.sol";
+import "@openzeppelinV2/contracts/token/ERC20/SafeERC20.sol";
+
+import "../../interfaces/yearn/IController.sol";
+import "../../interfaces/curve/Curve.sol";
+
+interface yvERC20 {
+    function deposit(uint256) external;
+
+    function withdraw(uint256) external;
+
+    function getPricePerFullShare() external view returns (uint256);
+}
+
+/*
+
+ A strategy must implement the following calls;
+ 
+ - deposit()
+ - withdraw(address) must exclude any tokens used in the yield - Controller role - withdraw should return to Controller
+ - withdraw(uint) - Controller | Vault role - withdraw should always return to vault
+ - withdrawAll() - Controller | Vault role - withdraw should always return to vault
+ - balanceOf()
+ 
+ Where possible, strategies must remain as immutable as possible, instead of updating variables, we update the contract by linking it in the controller
+ 
+*/
+
+contract StrategymUSDCurve {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public constant want = address(0xe2f2a5C287993345a840Db3B0845fbC70f5935a5);
+    address public constant mpool = address(0x8474DdbE98F5aA3179B3B3F5942D724aFcdec9f6);
+    address public constant m3crv = address(0x1AEf73d49Dedc4b1778d0706583995958Dc862e6);
+    address public constant yvm3crv = address(0x0FCDAeDFb8A7DfDa2e9838564c5A1665d856AFDF);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 1000;
```

Comment:
is this difference from the other strats expected?

https://github.com/iearn-finance/yearn-protocol/blob/2252c4f1f2af464dbf066da8e41a57498e4f6330/contracts/strategies/StrategyUSDC3pool.sol#L48

Reply:
Yes, it uses v2 params with v1 structure. This was about to be released on v2, but Andre said putting it on v1 was ideal.